src/ElasticaService.php

```diff
@@ -182,33 +180,29 @@ public function index($record)
 
         if (!$this->indexingMemorySet && $this->indexingMemory) {
             if ($this->indexingMemory == 'unlimited') {
-                ini_set('memory_limit', -1);
+                Environment::increaseMemoryLimitTo();
             } else {
-                ini_set('memory_limit', $this->indexingMemory);
+                Environment::increaseMemoryLimitTo($this->indexingMemory);
             }
             $this->indexingMemorySet = true;
         }
 
-        try {
-            $document = $record->getElasticaDocument();
-            $typeName = $record->getElasticaType();
-            $index = $this->getIndex();
+        $document = $record->getElasticaDocument();
```

Comment:
Indents don't look right

Reply:
Oh, you've remove the `try/catch`. Do none of the `$record` methods throw `Exception` instances?

Reply:
I can't remember the exact context, but I didn't want "create" actions etc... to silently fail. If my task errors when updating an index, I want it to throw all errors down the stack, even if we have a "logger", so my application can catch and respond to them as it feels fit. Catching exceptions too early makes app level error handling hard / impossible.